tt_metal/impl/device/device_mesh.hpp

```diff
@@ -39,6 +40,9 @@ class DeviceMesh
     Device *get_device(int row_idx, int col_idx) const;
     std::vector<Device *> get_devices_on_row(int row_idx) const;
     std::vector<Device *> get_devices_on_column(int col_idx) const;
+    std::vector<Device *> get_devices_on_ring() const;
+
+    void reorder_devices_to_ring();
```

Comment:
the plan is to remove this and reorder in the constructor in the next PR?

Reply:
Plan to me is to remove this, and use `get_devices_on_ring` instead of `get_devices` where that makes sense.